src/grid/chebyshev.jl

```diff
@@ -186,39 +244,48 @@ function chebdiff(n, x, f, invmat)
     return sum(intw .* f)
 end
 
-struct BaryCheb1D{N}
+struct BaryCheb1D{N,T}
```

Comment:
Shall we add a restriction like T<:AbstractFloat to aviod wrong inputs like Int?